plugin/cr/src/main/java/org/opencds/cqf/ruler/cr/KnowledgeArtifactProcessor.java

```diff
@@ -1250,14 +1254,87 @@ private void doesValueSetNeedExpansion(ValueSet vset, IFhirResourceDaoValueSet<V
 			vset.setExpansion(null);
 			ValueSetExpansionOptions options = new ValueSetExpansionOptions();
 			options.setIncludeHierarchy(true);
-			
+
 			ValueSet e = dao.expand(vset,options);
 			// we need to do this because dao.expand sets the expansion to a subclass and then that breaks the FhirPatch
 			// `copy` creates the superclass again
 			vset.setExpansion(e.getExpansion().copy());
 			return;
 		}
 	}
+
+	public void getExpansion(ValueSet valueSet) {
+		Extension authoritativeSource = valueSet
+			.getExtensionByUrl("http://hl7.org/fhir/StructureDefinition/valueset-authoritativeSource");
+		String urlToUse = authoritativeSource != null && authoritativeSource.hasValue() ? authoritativeSource.getValue().primitiveValue() : valueSet.getUrl();
+
+		ValueSet.ValueSetExpansionComponent expansion = new ValueSet.ValueSetExpansionComponent();
+		expansion.setTimestamp(Date.from(Instant.now()));
+
+		ValueSet expandedValueSet;
+		try {
+			expandedValueSet = terminologyServerClient.expand(valueSet, urlToUse);
+			valueSet.setExpansion(expandedValueSet.getExpansion());
+		} catch (Exception e) {
```

Comment:
Is there a more specific exception? Or better could the Client return a message if it fails instead of throwing?

Reply:
nvm didn't notice that it does still display the nestd 👍🏾  error